app/feature/feature-help-center/src/main/kotlin/com/hedvig/android/feature/help/center/ui/TextWithClickableDeepLink.kt

```diff
@@ -1,82 +0,0 @@
-package com.hedvig.android.feature.help.center.ui
-
-import android.content.Intent
-import android.net.Uri
-import androidx.compose.foundation.text.ClickableText
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
-import androidx.compose.ui.text.ExperimentalTextApi
-import androidx.compose.ui.text.SpanStyle
-import androidx.compose.ui.text.TextLayoutResult
-import androidx.compose.ui.text.TextStyle
-import androidx.compose.ui.text.buildAnnotatedString
-import androidx.compose.ui.text.style.TextDecoration
-import androidx.compose.ui.text.style.TextOverflow
-import androidx.compose.ui.text.withAnnotation
-import androidx.compose.ui.text.withStyle
-
-@OptIn(ExperimentalTextApi::class)
-@Composable
-internal fun TextWithClickableDeepLink(
-  text: String,
-  modifier: Modifier = Modifier,
-  style: TextStyle = TextStyle.Default,
-  softWrap: Boolean = true,
-  overflow: TextOverflow = TextOverflow.Clip,
-  maxLines: Int = Int.MAX_VALUE,
-  onTextLayout: (TextLayoutResult) -> Unit = {},
-  toDeepLink: (String) -> String,
-  linkStyle: SpanStyle = SpanStyle(
-    color = MaterialTheme.colorScheme.onBackground,
-    textDecoration = TextDecoration.Underline,
-  ),
-  linkMatcher: Regex = Regex("\\[([^\\[]+)](\\(.*\\))"),
-  onClick: (url: String) -> Unit = with(LocalContext.current) {
-    { url -> startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url))) }
-  },
-) {
-  val annotatedText = buildAnnotatedString {
-    val offset = linkMatcher.findAll(text).fold(0) { offset: Int, match: MatchResult ->
-      val linkText = match.groups[1]?.value
-      val url = match.groups[2]?.value?.removeSurrounding("(", ")")
```

Comment:
How is the link actually rendered now with the markdown library for the questions when the URL is there? Does it render just fine?

And yeah I see what you mean now, if we want to have the link works for dev/staging we'd have to take the string and manipulate it for those environments, probably with something very close to what we're doing here.
Ok to just let this be for now if this works anyway. We can sorta test that the links work in a way since the dev/staging build should deep link us into the real app on our device if we have it installed and that versions supports that link anyway.

Reply:
They render fine!